### ISSUE TITLE
Typescript autocomplete menu performance

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/feedback#622]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

A user on Twitter wrote:

> I'm currently using Zed as my main editor, but the TS autocomplete performance is not there yet. It's slow and by the time I decide to choose an option from the autocomplete, it changes and I select the wrong thing.
> 
> Why is the TS performance so slow?

This is something I've experience as well.  Not sure if its on our end of the language server end, but results appear super slow and you can see them being slowly filtered down, after you have finished typing.

### Expected behavior

-

### Environment

Zed 0.61.0 – /Applications/Zed.app
macOS 12.6
architecture x86_64

**Comment by iamnbutler** [comment]
Yeah typescript language server in general seems not great. I have to restart it a lot and get a LOT of phantom errors.

**Comment by morajabi** [comment]
What happens for me is updates to the auto complete dropdown come in a burst and I have to literally wait a while until it stops changing. Seems like it isn't debounced properly.

**Comment by devongovett** [comment]
Did anyone figure out what's going on here? In large TypeScript projects I've noticed that the language server is constantly at very high CPU usage, leading to very poor autocomplete performance, hangs in go to definition and hover, and out of date diagnostics. This is not the case in VSCode, so I'm guessing that it isn't TypeScript itself but somehow the way Zed is interfacing with it. Maybe duplicate requests or something, not sure.

If you want a reproduction, clone the [react-spectrum](https://github.com/adobe/react-spectrum) repo, run `yarn`, and then start editing some files. Everything is pretty laggy compared with VSCode.

I also noticed that it seems TypeScript specific, because the large Rust projects that I work on don't have this problem. Not sure if that helps.

**Comment by Aeolun** [comment]
I can confirm this. It's fairly aggravating to have finished typing the whole thing because autocomplete didn't show up, only to finally see it very slowly catch up after wards (I slowly see all the options I should've had before).
